src/Tools/dotnet-counters/Program.cs

```diff
@@ -135,6 +177,22 @@ internal class Program
                 Argument = new Argument<bool>(name: "resumeRuntime", getDefaultValue: () => true)
             };
 
+        private static Option MaxHistogramOption() =>
+            new Option(
+                alias: "--maxHistograms",
+                description: "The maximum number of histograms that can be tracked. Tracking more histograms uses more memory in the target process.")
+            {
+                Argument = new Argument<int>(name: "maxHistograms", getDefaultValue: () => 10)
+            };
+
+        private static Option MaxTimeSeriesOption() =>
+            new Option(
+                alias: "--maxTimeSeries",
+                description: "The maximum number of time series that can be tracked. Tracking more time series uses more memory in the target process.")
+            {
+                Argument = new Argument<int>(name: "maxTimeSeries", getDefaultValue: () => 1000)
```

Comment:
This also has a problem if we have a lot of tags.

Reply:
Thanks for the feedback - this sounds like it is mostly an issue of documentation though you also spotted a legitimate inconsistency with how the percentiles are being handled. 
1. The part that should be better documented - A "time series" refers to a single stream of measurements over time so it is by design that each unique set of tags maps to a different time series. This definition is also essential to the intended function of the time series limit, which is to prevent someone from accidentally triggering OutOfMemoryException in the target process solely by monitoring counters. For example imagine if I wrote the following code in my app:
```C#
myCounter.Add(1, KeyValuePair.Create("Time", DateTime.Now));
```
It is bad practice to write metrics where the dimensions (tags) have a very large number of possible values but with the monitoring off it wouldn't create any overt problem. As soon as dotnet-counters starts monitoring now every new tag value passed into this API creates a new allocation to store the measurement and that allocation is never freed until dotnet-counters disconnects. The time series limit puts a bound on those allocations, otherwise they would grow indefinitely until the process suffers an OOM.

2. Currently the 3 percentiles count as a single time series which does feel inconsistent with every other tag set which counts as 1. This is an issue in the MetricsEventSource itself because that is where the count of active time series is computed and enforced. I won't be able to fix it in this PR.

> This surfaces another issue I had with the tool: I can't ask to avoid tag aggregation.

It is a feature I wanted to add but didn't get time. I'd still like to add it in the future but we didn't get there in .NET 6 (unless you'd like to quickly implement it while I am on sabbatical? ; )

> This also has a problem if we have a lot of tags.

I think of the current design being neither the worst nor the best, but I didn't have time to implement better options this release. From worst to best in my POV:
1. Enable all metrics with no bound on tags -> apps that use too many tags crash with OOM
2. Bound the total time series but don't let the user pick which metrics to turn on -> user may see data they didn't care about and not see data they did care about
3. Bound total time series, user gets to pick metrics, but not tags -> user is more likely to see data they care about, but still possible they see unwanted tags
4. Bound total time series, user gets to pick metrics and tags -> user sees exactly what they wanted

We are at step 3 right now, step 4 would be ideal but we didn't get there (yet). Overall my hope is that the default time series limit is high enough that most people doing reasonable things will never hit it and they will never have reason to use the --maxTimeSeries argument. However I wanted to give people an escape valve because maybe my guess about how many time series is 'reasonable' is way off.

Reply:
I filed https://github.com/dotnet/runtime/issues/56901 to officially track the histogram time series accounting inconsistency, but then resolved it "Won't Fix" and closed it because I doubt anyone is going to notice or care unless they were explicitly trying to probe this behavior.